milli/src/fields_ids_map.rs

```diff
@@ -41,6 +41,16 @@ impl FieldsIdsMap {
         }
     }
 
+    /// Get the ids of a field and all its nested fields based on its name.
+    pub fn nested_ids(&self, name: &str) -> Vec<FieldId> {
+        self.names_ids
+            .range(name.to_string()..)
+            .take_while(|(key, _)| key.starts_with(name))
+            .filter(|(key, _)| crate::is_faceted_by(key, name))
+            .map(|(_name, id)| *id)
+            .collect()
+    }
```

Comment:
meta note but all this string manipulation makes me wonder if we wouldn't want to store nested fields as arrays rather than as dot separated strings?

Reply:
Can you give an example? I’m not sure to understand what you would put in it?

Reply:
instead of storing nested fields as `"root.grandparent.parent.field"`, we would store them as `["root", "grandparent", "parent", "field"]` (or probably, more accurately, with an array of the field ids).

Then `is_faceted_by(field)` becomes `path[..path.len()-1].contains(field)` and `nested_ids(name)` becomes `names_ids.filter(|path| path[0] == name)`

Reply:
not saying we should change this in this PR of course!